### PR TITLE
chore: Update to newer argocd version for better API compatibility

### PR DESCRIPTION
**Pull request by jessebye**
Should resolve bugs like this:
- https://github.com/argoproj-labs/argocd-image-updater/issues/593
- https://github.com/argoproj-labs/argocd-image-updater/issues/554

That are caused by using an outdated version of Argo's Application spec.

### CODE REVIEW

**Comment by jessebye** [comment]
@jannfis If you would kindly review? This should resolve at least two reported bugs and unblock our team as well! Thank you!

**Comment by jessebye** [comment]
Built and pushed to Dockerhub for testing. Works great in our environment, resolves the issue we were having with some fields being removed every time it updated the Argo Applications. https://hub.docker.com/r/riskalyze/argocd-image-updater

File: pkg/argocd/argocd_test.go [hunk 1]
```diff
@@ -188,7 +188,7 @@ func Test_FilterApplicationsForUpdate(t *testing.T) {
 				},
 				Spec: v1alpha1.ApplicationSpec{},
 				Status: v1alpha1.ApplicationStatus{
-					SourceType: v1alpha1.ApplicationSourceTypeKsonnet,
+					SourceType: v1alpha1.ApplicationSourceTypePlugin,
```
Comment (jessebye) [inline pkg/argocd/argocd_test.go:191]: Ksonnet was deprecated and removed from Argo CD, so we need to use a different unsupported source type here.

File: go.mod [hunk 2]
```diff
@@ -4,34 +4,34 @@ go 1.20
 
 require (
 	github.com/Masterminds/semver v1.5.0
-	github.com/argoproj/argo-cd/v2 v2.2.7
-	github.com/argoproj/gitops-engine v0.5.2
-	github.com/argoproj/pkg v0.11.1-0.20211203175135-36c59d8fafe0
+	github.com/argoproj/argo-cd/v2 v2.7.7
```
Comment (jessebye) [inline go.mod:7]: Updates argo-cd and associated modules to latest matching versions.

**Comment by jannfis** [comment]
Thanks a lot for updating the PR, @jessebye, I very much appreciate it!

Please allow me some time for review & merge, as I'm off this week. I will come back to this PR as soon as time allows.

**Review by jannfis — APPROVED** [review]
LGTM, thanks a lot!

Sorry for taking a while to review & merge.

**Pull request merged**